Core/include/Acts/TrackFinding/MeasurementSelector.ipp

```diff
@@ -0,0 +1,156 @@
+
+  double minChi2 = std::numeric_limits<double>::max();
+  std::size_t minIndex = std::numeric_limits<std::size_t>::max();
+
+  isOutlier = false;
+
+  // Loop over all measurements to select the compatible measurements
+  // Sort track states which do not satisfy the chi2 cut to the end.
+  // When done trackStateIterEnd will point to the first element that
+  // does not satisfy the chi2 cut.
+  std::size_t passedCandidates = 0ul;
+  for (std::size_t i(0ul); i < candidates.size(); ++i) {
+    auto& trackState = candidates[i];
+
+    // This abuses an incorrectly sized vector / matrix to access the
+    // data pointer! This works (don't use the matrix as is!), but be
+    // careful!
+    double chi2 = calculateChi2(
+        trackState
+            .template calibrated<MultiTrajectoryTraits::MeasurementSizeMax>()
+            .data(),
+        trackState
+            .template calibratedCovariance<
+                MultiTrajectoryTraits::MeasurementSizeMax>()
+            .data(),
+        trackState.predicted(), trackState.predictedCovariance(),
+        trackState.projector(), trackState.calibratedSize());
+    trackState.chi2() = chi2;
+
+    if (chi2 < minChi2) {
+      minChi2 = chi2;
+      minIndex = i;
+    }
+
+    // only consider track states which pass the chi2 cut
+    if (chi2 >= maxChi2Cut) {
+      continue;
+    }
+
+    if (passedCandidates != i) {
+      std::swap(candidates[passedCandidates], candidates[i]);
+      if (minIndex == i)
+        minIndex = passedCandidates;
+    }
+    ++passedCandidates;
+  }
+
+  // If there are no measurements below the chi2 cut off, return the
+  // measurement with the best chi2 and tag it as an outlier
+  if (passedCandidates == 0ul) {
+    ACTS_VERBOSE("No measurement candidate. Return an outlier measurement chi2="
+                 << minChi2);
+    isOutlier = true;
+  }
+
+  if (passedCandidates <= 1ul || numMeasurementsCut == 1ul) {
+    // return single item range, no sorting necessary
+    ACTS_VERBOSE("Returning only 1 element");
+    return Result::success(std::make_pair(candidates.begin() + minIndex,
+                                          candidates.begin() + minIndex + 1));
+  }
+
+  std::sort(
+      candidates.begin(), candidates.begin() + passedCandidates,
+      [](const auto& tsa, const auto& tsb) { return tsa.chi2() < tsb.chi2(); });
+
+  if (passedCandidates <= numMeasurementsCut) {
+    ACTS_VERBOSE("Number of selected measurements: "
+                 << passedCandidates << ", max: " << numMeasurementsCut);
+    return Result::success(std::make_pair(
+        candidates.begin(), candidates.begin() + passedCandidates));
+  }
+
+  ACTS_VERBOSE("Number of selected measurements: "
+               << numMeasurementsCut << ", max: " << numMeasurementsCut);
+
+  return Result::success(std::make_pair(
+      candidates.begin(), candidates.begin() + numMeasurementsCut));
+}
+
+template <typename traj_t, typename cut_value_t>
+cut_value_t MeasurementSelector::VariableCut(
```

Comment:
this was there before but since you are touching git: can we rename this to something like `getCut`